### PR TITLE
Simplify `DynamicType` binary op definitions with `dispatch`

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: lib/dynamic_type/test/assignment.cpp [hunk 1]
```diff
@@ -13,28 +13,27 @@
...
-            ::testing::HasSubstr("Cannot compute ")));     \
-    EXPECT_THAT(                                           \
-        []() {                                             \
-          IntSomeType x(SomeType{});                       \
-          x += 1;                                          \
-        },                                                 \
-        ::testing::ThrowsMessage<std::runtime_error>(      \
-            ::testing::HasSubstr("Cannot compute ")));     \
-    static_assert(!(opcheck<SomeTypes&> += opcheck<int>)); \
+#define TEST_ASSIGN_OP(op, assign_op, name)                                \
```
Comment (zasdfgbnm) [inline lib/dynamic_type/test/assignment.cpp:16]: Just changing the error message

File: lib/dynamic_type/test/binary_ops.cpp [hunk 2]
```diff
@@ -66,21 +66,20 @@
         (3L op 2L));                                                           \
     EXPECT_THAT(                                                               \
         [&]() { DoubleInt64Bool() op DoubleInt64Bool(2L); },                   \
-        ::testing::ThrowsMessage<std::runtime_error>(                          \
-            ::testing::HasSubstr("Cannot compute ")));                         \
+        ::testing::ThrowsMessage<std::runtime_error>(::testing::HasSubstr(     \
```
Comment (zasdfgbnm) [inline lib/dynamic_type/test/binary_ops.cpp:69]: Just changing the error message

File: lib/dynamic_type/test/utils.h [hunk 3]
```diff
@@ -51,5 +51,3 @@ struct DoubleInt64BoolVecTwo {
 };
 using IntSomeType = DynamicType<NoContainers, int, SomeType, NonInstantiable>;
 using BoolSomeType = DynamicType<NoContainers, bool, SomeType, NonInstantiable>;
-using SomeTypes =
```
Comment (zasdfgbnm) [inline lib/dynamic_type/test/utils.h:54]: Now we are assuming that types in the DynamicType only appears once, I believe this is a fair assumption.

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**